# Rainfall in Durham

**Instructor solution.** Run this notebook independently from top to bottom.

In [1]:
from pathlib import Path
import sys

# Support kernels started in the repository root or in instructor-notebooks.
ROOT = Path.cwd()
if not (ROOT / "data").is_dir():
    ROOT = ROOT.parent
if not (ROOT / "data").is_dir():
    raise FileNotFoundError("Open this notebook from the repository or instructor-notebooks folder.")
sys.path.insert(0, str(ROOT))

These fictional measurements record rainfall in millimeters for seven consecutive days in Durham.

In [2]:
location = "Durham"
rainfall_mm = [0.0, 4.2, 12.5, 0.0, 2.1, 18.3, 0.0]

In [3]:
total_rainfall = sum(rainfall_mm)
measurement_count = len(rainfall_mm)
average_rainfall = total_rainfall / measurement_count
largest_rainfall = max(rainfall_mm)
third_day = rainfall_mm[2]
print("Location:", location)
print("Total (mm):", total_rainfall)
print("Measurements:", measurement_count)
print("Average (mm/day):", average_rainfall)
print("Largest (mm):", largest_rainfall)
print("Third day (mm):", third_day)

Location: Durham
Total (mm): 37.1
Measurements: 7
Average (mm/day): 5.3
Largest (mm): 18.3
Third day (mm): 12.5


The total is 37.1 mm, the mean is 5.3 mm/day, and the largest daily measurement is 18.3 mm. The third day is index 2 and has 12.5 mm. Include dry days when calculating the daily mean.